src/components/LeaderBoard/Leaderboard.jsx

```diff
@@ -127,30 +211,83 @@ function LeaderBoard({
   };
 
+  const TeamSelected = team => {
+    if (team) {
+      const teamName = `${team.teamName.substring(0, 15)}...`;
+      setSelectedTeamName(team.teamName.length >= 15 ? teamName : team.teamName);
+      setUsersSelectedTeam(team);
+    } else {
+      setSelectedTeamName('View All');
+      setUsersSelectedTeam(null);
+    }
+  };
+
   return (
     <div>
-      <h3>
-        <div className="d-flex align-items-center">
-          <span className="mr-2">Leaderboard</span>
-          <i
-            data-toggle="tooltip"
-            data-placement="right"
-            title="Click to refresh the leaderboard"
-            style={{ fontSize: 24, cursor: 'pointer' }}
-            aria-hidden="true"
-            className={`fa fa-refresh ${isLoading ? 'animation' : ''}`}
-            onClick={updateLeaderboardHandler}
-          />
-          &nbsp;
-          <EditableInfoModal
-            areaName="LeaderboardOrigin"
-            areaTitle="Leaderboard"
-            role={loggedInUser.role}
-            fontSize={24}
-            isPermissionPage
-          />
-        </div>
-      </h3>
+      <div>
+        <h3>
+          <div className="d-flex align-items-center">
+            <span className="mr-2">Leaderboard</span>
+            <i
+              data-toggle="tooltip"
+              data-placement="right"
+              title="Click to refresh the leaderboard"
+              style={{ fontSize: 24, cursor: 'pointer' }}
+              aria-hidden="true"
+              className={`fa fa-refresh ${isLoading ? 'animation' : ''}`}
+              onClick={updateLeaderboardHandler}
+            />
+            &nbsp;
+            <EditableInfoModal
+              areaName="LeaderboardOrigin"
+              areaTitle="Leaderboard"
+              role={loggedInUser.role}
+              fontSize={24}
+              isPermissionPage
+            />
+          </div>
+        </h3>
+
+        {userRole === 'Administrator' || userRole === 'Owner' ? (
+          <section className="d-flex flex-row flex-wrap mb-3">
+            <Dropdown isOpen={dropdownOpen} toggle={toggleDropdown} className=" mr-3">
+              <DropdownToggle caret>
+                {selectedTeamName || 'Select a Team'} {/* Display selected team or default text */}
+              </DropdownToggle>
+              <DropdownMenu>
+                {teams.length === 0 ? (
+                  <DropdownItem
+                    onClick={() => toast.warning('Please, create a team to use the filter.')}
+                  >
+                    Please, create a team to use the filter.
+                  </DropdownItem>
+                ) : (
+                  /* eslint-disable */
+                  <>
+                    <DropdownItem onClick={() => TeamSelected(null)}>{'View All'}</DropdownItem>
+                    {teams.map(team => (
+                      <DropdownItem key={team._id} onClick={() => TeamSelected(team)}>
+                        {team.teamName}
```

Comment:
Hi, I've tested this feature and my observations are as below:
What works:
- Without adding team, the 'My Team' was not selectable as expected.
- Able to add team, add members, filter based on 'My Team' and 'View All' in dashboard.
- functionality of  filter with buttons for 1 day, 2 days and so forth works.
- Other users like CoreTeam/Volunteer can't see Team filter as expected.

**Issue**:
- When a team name with large characters is added, the same is reflected in the dashboard which disrupts the team selection view. (Refer Issue screenshots below at the end)

**Suggestion**: On line 270, recommend using `${team.teamName.substring(0, 10)}...` in the DropdownItem in the Leaderboard and under Team member task in dashboard page.

Other observations:
- After filtering based on a team in the leaderboard, when we hit refresh the list shows all users of the HGN software, even though a particular team is selected. (Please confirm if this is the expected functionality, screenshot attached)


![1 Created Team](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78548214/51e083f5-cfc8-4547-b752-5f2e8636d024)
![2 assigned team members](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78548214/84b8fad2-5e44-4912-8f95-b5bd827a49ea)
![3 Team filtered in dashboard](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78548214/c61458bc-c1df-4d3d-96d7-08b20edf317b)
![4 viewAll expands  team](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78548214/458840c3-80d9-4a41-b2bf-2e4f74b13d27)
![5 CoreTeam](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78548214/f3c982c4-d5e6-468d-b974-25db1fd33f31)


**Issue screenshots:**
![Issue - Team length 2](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78548214/b0f83ccc-88dd-489f-af57-550ce2828f26)
![Issue - team length 3](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78548214/7393528a-fb15-437b-91d1-c13a298be57f)
![Issue - Team length](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78548214/ccdc5d74-a292-407a-bfb1-bf8bbeeb568a)


Observation on leaderboard refresh:
![Observ - leaderboard update with team selection](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78548214/9420dbc9-5d87-4dd6-8aab-ba9132b765dc)